activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -120,12 +125,16 @@ def initialize(*addresses)
           raise ArgumentError, "First argument must be an empty array, an array of hosts or a Dalli::Client instance."
         end
         if addresses.first.is_a?(Dalli::Client)
+          ActiveSupport.deprecator.warn(<<~MSG)
+            Initializing MemCacheStore with a Dalli::Client is deprecated and will be removed in Rails 7.2.
+            Please pass an address instead.
```

Comment:
```suggestion
            Use memcached server addresses instead.
```

This echoes the language in the first sentence of the docs.